source/codegen/templates/service_helpers.mako

```diff
@@ -65,8 +65,9 @@ ${set_response_values(output_parameters)}\
 %>\
 ${initialize_input_params(function_name, parameters)}\
 ${initialize_output_params(output_parameters)}\
-% if function_name == config['close_function']:
-      session_repository_->remove_session(${service_helpers.create_args(parameters)});
+% if function_name == config['close_function'] or service_helpers.is_custom_close_method(function_data):
+      session_repository_->remove_session(${service_helpers.create_args(parameters[:1])});
+      library_->${function_name}(${service_helpers.create_args(parameters)});
```

Comment:
This is something we did contemplate upon when we decided to go with feature branches approach. Our thought process was, since both features would be making independent changes into driver-specific files we shouldn't run into many conflicts. We thought any major infrastructure changes we make, we could directly bowl into the public repo and then merge down to feature branches. But that turned out to be too much overhead and making such changes alongwith the features that needed them was more convenient and meaningful. We ran into handful of situations where we had to make similar changes in both feature branches (3, IIRC).

The reason for feature branches was so as to be able to release individual drivers as and when they are ready. Given that we are really planning at iteration level and any drivers we start working on in parallel would probably finish within a week or two of each other, I think it should be ok to create branches per iteration to avoid such issues. @harsha-bhushan  @ganeshnie  thoughts?

Reply:
I agree. branch per iteration is acceptable.

Reply:
@jonlugoatx as far as Kruthi's PR for making same change in dcpower branch is considered, we are planning to merge one feature branch into main, run system tests and also validate examples against that. And then merge that back with second feature branch, do the same testing again to make sure that no bugs are introduced by the merge. Feature validation will happen in public repo, so that should take care of discovering any bugs. Given that this is mostly last piece of work pending for both the features, I think we can make this change in both branches independently and follow single branch approach I mentioned in my earlier comment to address this issue in future.

Reply:
That sounds reasonable to me. My opinion is that we should be following the mainline model where all changes are made in main and any features we introduce we "toggle off" somehow until they're ready. Maybe at some point it would be useful to think through how such a toggle could be implemented, but for now I do not want to disrupt your process.

Reply:
Feature toggles was one of the options on table as well. That was ruled out because of timing constraints. But I agree that it makes sense to invest in that, since we will have more drivers integrating into our repo soon, and mainline model will definitely help in long run.